File: backend/src/main/java/com/marmouset/workout/app/workout/usecase/impl/ListWorkoutLogsUseCaseImpl.java
package com.marmouset.workout.app.workout.usecase.impl;

import com.marmouset.workout.app.workout.entity.WorkoutLog;
import com.marmouset.workout.app.workout.usecase.ListWorkoutLogsUseCase;
import com.marmouset.workout.app.workout.usecase.WorkoutLogRepository;
import java.util.List;

/**
 * Use case for listing the workout logs.
 */
public class ListWorkoutLogsUseCaseImpl implements ListWorkoutLogsUseCase {
  private final WorkoutLogRepository repository;

  /**
   * Constructor for this use case.
   *
   * @param repository the workout logContainer repository
   */
  public ListWorkoutLogsUseCaseImpl(WorkoutLogRepository repository) {
    this.repository = repository;
  }


  @Override
  public List<? extends WorkoutLog> list() {
    return repository.read();
  }
}

File: frontend/src/app/workout/usecase/impl/ListWorkoutLogsUseCaseImpl.ts
import WorkoutLog from '../../entity/WorkoutLog';
import ListWorkoutLogsUseCase from '../ListWorkoutLogsUseCase';
import WorkoutClientGateway from '../../adapter/WorkoutClientGateway';

export default class ListWorkoutLogsUseCaseImpl 
implements ListWorkoutLogsUseCase {
  constructor(
    private readonly clientGateway: WorkoutClientGateway
  ) {}

  async list(): Promise<WorkoutLog[]> {
    return this.clientGateway.list();
  }
}
